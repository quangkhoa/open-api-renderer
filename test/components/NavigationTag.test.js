import React from 'react'
import NavigationTag from './../../src/components/NavigationTag/NavigationTag'
import renderer from 'react-test-renderer'

describe('<NavigationTag />', () => {
  const hash = '#pets'

  it('renders with minimum props', () => {
    const tree = renderer.create(
      <NavigationTag title='pets' onClick={jest.fn()} hash={hash} />
    )

    expect(tree).toMatchSnapshot()
  })

  it('renders with methods', () => {
    const methods = [{
      type: 'get',
      title: 'Get pets',
      link: 'pets'
    },
    {
      type: 'get',
      title: 'Get pets',
      link: 'pets/:id'
    }]

    const tree = renderer.create(
      <NavigationTag title='pets' onClick={jest.fn()} hash={hash} methods={methods} />
    )

    expect(tree).toMatchSnapshot()
  })

  it('renders with methods having paths', () => {
    const methods = [{
      type: 'get',
      title: 'Get pets',
      path: '/pets',
      link: '/pets/get'
    },
    {
      type: 'get',
      title: 'Get pets',
      path: '/pets/:id',
      link: '/pets/:id/get'
    }]

    const tree = renderer.create(
      <NavigationTag title='pets' onClick={jest.fn()} hash={hash} methods={methods} navigationMethodDisplayType={'path'} />
    )

    expect(tree).toMatchSnapshot()
  })

  it('renders with methods having paths and summary', () => {
    const methods = [{
      type: 'get',
      title: 'Get pets',
      path: '/pets',
      link: '/pets/get'
    },
    {
      type: 'get',
      title: 'Get pets',
      path: '/pets/:id',
      link: '/pets/:id/get'
    }]

    const tree = renderer.create(
      <NavigationTag title='pets' onClick={jest.fn()} hash={hash} methods={methods} navigationMethodDisplayType={'all'} />
    )

    expect(tree).toMatchSnapshot()
  })

  it('renders with description', () => {
    const tree = renderer.create(
      <NavigationTag title='Pets' handle='pets' description='Access to Pets' onClick={jest.fn()} hash={hash} />
    )

    expect(tree).toMatchSnapshot()
  })
})
