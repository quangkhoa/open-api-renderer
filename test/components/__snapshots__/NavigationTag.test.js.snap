// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavigationTag /> renders with description 1`] = `
<div>
  <a
    className={undefined}
    href="#Pets"
    onClick={[Function]}
  >
    <span>
      Pets
    </span>
    <div
      className=""
    >
      <img
        alt=""
        src="test-file-stub"
        title="arrow"
      />
    </div>
    <div
      className=""
      dangerouslySetInnerHTML={
        Object {
          "__html": "<p>Access to Pets</p>
",
        }
      }
    />
  </a>
  <div
    className=""
  />
</div>
`;

exports[`<NavigationTag /> renders with methods 1`] = `
<div>
  <a
    className={undefined}
    href="#pets"
    onClick={[Function]}
  >
    <span>
      pets
    </span>
    <div
      className=""
    >
      <img
        alt=""
        src="test-file-stub"
        title="arrow"
      />
    </div>
  </a>
  <div
    className="undefined"
  >
    <a
      className=""
      href="#pets"
      onClick={undefined}
    >
      <span
        className="get"
      >
        GET
      </span>
      <span
        className="short"
      >
        Get pets
      </span>
    </a>
    <a
      className=""
      href="#pets/:id"
      onClick={undefined}
    >
      <span
        className="get"
      >
        GET
      </span>
      <span
        className="short"
      >
        Get pets
      </span>
    </a>
  </div>
</div>
`;

exports[`<NavigationTag /> renders with methods having paths 1`] = `
<div>
  <a
    className={undefined}
    href="#pets"
    onClick={[Function]}
  >
    <span>
      pets
    </span>
    <div
      className=""
    >
      <img
        alt=""
        src="test-file-stub"
        title="arrow"
      />
    </div>
  </a>
  <div
    className=""
  >
    <a
      className="undefined"
      href="#/pets/get"
      onClick={undefined}
    >
      <span
        className="get"
      >
        GET
      </span>
      <span
        className={undefined}
      >
        /pets
      </span>
    </a>
    <a
      className="undefined"
      href="#/pets/:id/get"
      onClick={undefined}
    >
      <span
        className="get"
      >
        GET
      </span>
      <span
        className={undefined}
      >
        /pets/:id
      </span>
    </a>
  </div>
</div>
`;

exports[`<NavigationTag /> renders with methods having paths and summary 1`] = `
<div>
  <a
    className={undefined}
    href="#pets"
    onClick={[Function]}
  >
    <span>
      pets
    </span>
    <div
      className=""
    >
      <img
        alt=""
        src="test-file-stub"
        title="arrow"
      />
    </div>
  </a>
  <div
    className=""
  >
    <a
      className="undefined"
      href="#/pets/get"
      onClick={undefined}
    >
      <span
        className="get"
      >
        GET
      </span>
      <span
        className={undefined}
      >
        /pets
      </span>
      <span
        className=""
      >
        Get pets
      </span>
    </a>
    <a
      className="undefined"
      href="#/pets/:id/get"
      onClick={undefined}
    >
      <span
        className="get"
      >
        GET
      </span>
      <span
        className={undefined}
      >
        /pets/:id
      </span>
      <span
        className=""
      >
        Get pets
      </span>
    </a>
  </div>
</div>
`;

exports[`<NavigationTag /> renders with minimum props 1`] = `
<div>
  <a
    className={undefined}
    href="#pets"
    onClick={[Function]}
  >
    <span>
      pets
    </span>
    <div
      className=""
    >
      <img
        alt=""
        src="test-file-stub"
        title="arrow"
      />
    </div>
  </a>
  <div
    className=""
  />
</div>
`;
